### PR TITLE
[ISSUE-40] - Tidy Resources More Aggressively 

### PR DESCRIPTION
**Pull request by mtjandra**
* Addresses https://github.com/chef/inspec-aws/issues/40
* When running `bundle exec rake test:integration` cleanup resources even if integration tests fail
* This should help reduce orphaned resources
* The issue hinted at passing in a parameter to optionally leave resources hanging, but I feel if you want to be debugging with active resources, you should run the separate ruby tasks on your own
* I tested by removing the ec2 setup from my Terraform build and running the integration tests as is. Below is the sample output near the end.

```

Profile: inspec-aws-integration-tests
Version: (not specified)
Target:  local://


  EC2 Instance
     ∅  undefined method `exists?' for nil:NilClass
     ∅  undefined method `image_id' for nil:NilClass
     ∅  undefined method `instance_type' for nil:NilClass
     Did you mean?  instance_exec
                    instance_eval
                    instance_of?
  EC2 Instance
     ∅  Invalid id: "default.example_ec2_id"
     ∅  Invalid id: "default.example_ec2_id"
     ∅  Invalid id: "default.example_ec2_id"
  EC2 Instance
     ✔  i-06b4bc106e0d03dfd should not exist
  IAM Access-Key
     ✔  not-an-id should not exist
  aws_iam_user has_mfa_enabled?
     ✔  should equal false
     ✔  should equal false
  aws_iam_user has_console_password?
     ✔  should equal true


Profile: InSpec AWS Resource Pack (inspec-aws)
Version: 1.0.0
Target:  local://

     No tests executed.

Test Summary: 5 successful, 6 failures, 0 skipped
Integration testing has failed
----> Cleanup
cd /home/mtjandra/code/inspec-aws/test/integration/build/ && terraform destroy -force
aws_iam_user.mfa_not_enabled_user: Refreshing state... (ID: tjandrawidjaja.mfa_not_enabled_user)
aws_iam_user.console_password_enabled_user: Refreshing state... (ID: tjandrawidjaja.console_password_enabled_user)
aws_iam_user_login_profile.u: Refreshing state... (ID: tjandrawidjaja.console_password_enabled_user)
aws_iam_user_login_profile.u: Destroying... (ID: tjandrawidjaja.console_password_enabled_user)
aws_iam_user.mfa_not_enabled_user: Destroying... (ID: tjandrawidjaja.mfa_not_enabled_user)
aws_iam_user_login_profile.u: Destruction complete
aws_iam_user.console_password_enabled_user: Destroying... (ID: tjandrawidjaja.console_password_enabled_user)
aws_iam_user.mfa_not_enabled_user: Destruction complete
aws_iam_user.console_password_enabled_user: Destruction complete

Destroy complete! Resources: 3 destroyed.
----> Destroying terraform environment
cd /home/mtjandra/code/inspec-aws/test/integration/build/ && terraform env select default
Switched to environment "default"!
cd /home/mtjandra/code/inspec-aws/test/integration/build && terraform env delete tjandrawidjaja
Deleted environment "tjandrawidjaja"!
[inspec-aws (issue40)]> echo $?
1
```

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -70,11 +70,15 @@ namespace :test do
...
-    Rake::Task["test:setup_integration_tests"].execute
-    Rake::Task["test:run_integration_tests"].execute
-    Rake::Task["test:cleanup_integration_tests"].execute
-    Rake::Task["test:destroy_test_environment"].execute
+    begin
+      Rake::Task["test:configure_test_environment"].execute    
+      Rake::Task["test:setup_integration_tests"].execute
+      Rake::Task["test:run_integration_tests"].execute
+    rescue
+      abort("Integration testing has failed")
```
Comment (mtjandra) [inline Rakefile]: Used to maintain exit code of 1

**Review by Rugbyte — APPROVED** [review]

File: Rakefile (hunk 1 above)
Comment (chris-redekop) [inline Rakefile]: Nice.  This is easier than I thought it would be!

**Review by chris-redekop — APPROVED** [review]

**Review by AlexBedley — APPROVED** [review]

**Comment by chris-redekop** [comment]
Any feedback on this one, @Stiffni ?

**Review by Stiffni — APPROVED** [review]

**Review by chris-rock — APPROVED** [review]
@mtjandra Great cleanup. Thank you!

**Pull request merged**